sklearn/ensemble/_gb.py

```diff
@@ -139,6 +135,20 @@ def update(self, j, est):
 class BaseGradientBoosting(BaseEnsemble, metaclass=ABCMeta):
     """Abstract base class for Gradient Boosting."""
 
+    _parameter_constraints = {
+        **DecisionTreeRegressor._parameter_constraints,
+        "learning_rate": [Interval(Real, 0.0, None, closed="neither")],
```

Comment:
Similarly to `tol`, we theoretically can allow `learning_rate=0` which could be useful in debugging contexts.
```suggestion
        "learning_rate": [Interval(Real, 0.0, None, closed="left")],
```